server/routes/punishments/activateSuspendedPunishments/activateSuspendedPunishmentsPage.ts

```diff
@@ -0,0 +1,51 @@
+import url from 'url'
+import { Request, Response } from 'express'
+import { FormError } from '../../../@types/template'
+
+import UserService from '../../../services/userService'
+import adjudicationUrls from '../../../utils/urlGenerator'
+import { hasAnyRole } from '../../../utils/utils'
+import PunishmentsService from '../../../services/punishmentsService'
+
+export default class ActivateSuspendedPunishmentsPage {
+  constructor(private readonly punishmentsService: PunishmentsService, private readonly userService: UserService) {}
+
+  private renderView = async (req: Request, res: Response, error: FormError | null): Promise<void> => {
+    const adjudicationNumber = Number(req.params.adjudicationNumber)
+    const { user } = res.locals
+    const suspendedPunishmentDetails = await this.punishmentsService.getSuspendedPunishmentDetails(
+      adjudicationNumber,
+      user
+    )
+    return res.render(`pages/activateSuspendedPunishments.njk`, {
+      awardPunishmentsHref: adjudicationUrls.awardPunishments.urls.modified(adjudicationNumber),
+      manuallyActivateSuspendedPunishmentsHref:
+        adjudicationUrls.manuallyActivateSuspendedPunishment.urls.start(adjudicationNumber),
+      prisonerName: suspendedPunishmentDetails.prisonerName,
+      suspendedPunishments: suspendedPunishmentDetails.suspendedPunishments,
+      errors: error ? [error] : [],
+    })
+  }
+
+  view = async (req: Request, res: Response): Promise<void> => {
+    const userRoles = await this.userService.getUserRoles(res.locals.user.token)
+    if (!hasAnyRole(['ADJUDICATIONS_REVIEWER'], userRoles)) {
+      return res.render('pages/notFound.njk', { url: req.headers.referer || adjudicationUrls.homepage.root })
+    }
+
+    return this.renderView(req, res, null)
+  }
+
+  submit = async (req: Request, res: Response): Promise<void> => {
+    const adjudicationNumber = Number(req.params.adjudicationNumber)
+    const { activate } = req.body
+    const punishmentToActivate = activate.split('-').slice(-1)[0] || null
```

Comment:
Looking at `activateSuspendedPunishments.test.ts` I think this grabs the punishment number? In that test we pull out `71` from `suspended-punishment-71`, could name the variable to something like `punishmentNumberToActivate` - but I might be finding it confusing because I don't know the domain too well 😄